### PR TITLE
ENH: Add More Resampling Modes To Resample Image Geometry

### PR DESCRIPTION
**Pull request by joeykleingers**
+ Added "Scaling" and "Exact Dimensions" modes along with the default "Spacing" mode.

NOTE: ResampleImageGeom_Exemplar.tar.gz must be removed from the DataArchive after this PR is merged in!

### CODE REVIEW

**Comment by joeykleingers** [comment]
NOTE: ResampleImageGeom_Exemplar.tar.gz must be removed from the DataArchive after this PR is merged in!

File: src/Plugins/ComplexCore/docs/ResampleImageGeomFilter.md [hunk 1]
```diff
@@ -6,19 +6,35 @@ Sampling (Resample)
...
 ## Description ##
 
-This **Filter** changes the **Cell** spacing/resolution based on inputs from the user. The values entered are the desired new spacings (not multiples of the current resolution).  The number of **Cells** in the volume will change when the spacing values are changed and thus the user should be cautious of generating "too many" **Cells** by entering very small values (i.e., very high resolution). Thus, this **Filter** will perform a down-sampling or up-sampling procedure.  
+This **Filter** changes the **Cell** spacing/resolution based on inputs from the user. There are several resampling modes:
+
+### Spacing ###
+The values entered are the desired new spacings (not multiples of the current resolution).  The number of **Cells** in the volume will change when the spacing values are changed and thus the user should be cautious of generating "too many" **Cells** by entering very small values (i.e., very high resolution).
+
+### Scale Factor ###
+The values entered are the desired scaling factor for each dimension, in percentages.  (50%, 50%, 50%) would resample the geometry by half in all dimensions.
```
Comment (imikejackson) [inline src/Plugins/ComplexCore/docs/ResampleImageGeomFilter.md]: Can you give a more concrete example by stating for a given scaling and dimensions, if I put in 50% in all directions, what do I end up with for dimensions and scaling.

File: src/Plugins/ComplexCore/docs/ResampleImageGeomFilter.md [hunk 2]
```diff
@@ -6,19 +6,35 @@ Sampling (Resample)
 
 ## Description ##
 
-This **Filter** changes the **Cell** spacing/resolution based on inputs from the user. The values entered are the desired new spacings (not multiples of the current resolution).  The number of **Cells** in the volume will change when the spacing values are changed and thus the user should be cautious of generating "too many" **Cells** by entering very small values (i.e., very high resolution). Thus, this **Filter** will perform a down-sampling or up-sampling procedure.  
+This **Filter** changes the **Cell** spacing/resolution based on inputs from the user. There are several resampling modes:
+
+### Spacing ###
+The values entered are the desired new spacings (not multiples of the current resolution).  The number of **Cells** in the volume will change when the spacing values are changed and thus the user should be cautious of generating "too many" **Cells** by entering very small values (i.e., very high resolution).
```
Comment (imikejackson) [inline src/Plugins/ComplexCore/docs/ResampleImageGeomFilter.md:12]: Give a concrete example using example dimensions and scaling.

File: src/Plugins/ComplexCore/docs/ResampleImageGeomFilter.md [hunk 3]
```diff
@@ -6,19 +6,35 @@ Sampling (Resample)
...
+
+### Scale Factor ###
+The values entered are the desired scaling factor for each dimension, in percentages.  (50%, 50%, 50%) would resample the geometry by half in all dimensions.
+
+### Exact Dimensions ###
+The values entered are the desired cell dimensions of the resampled geometry.  (100, 100, 100) would resample the geometry so that there are 100 cells in each dimension.
```
Comment (imikejackson) [inline src/Plugins/ComplexCore/docs/ResampleImageGeomFilter.md:42]: How would this effect the scaling values in each direction. Give a concrete example.

**Review by imikejackson — CHANGES_REQUESTED** [review]
Can you also create an example pipeline that shows the usage of each kind of resampling. Just use one of the "Porosity" images and an example. Does not have to be fancy image.

**Review by imikejackson — APPROVED** [review]

**Pull request merged**